providers/AuthProvider.tsx

```diff
@@ -0,0 +1,202 @@
+import { useRouter } from "next/router";
+import {
+  createContext,
+  ReactNode,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import useSWR from "swr";
+import {
+  getUser,
+  getUserSWR,
+  signInWithEmailAndPassword,
+  signInWithGoogle,
+  signOut,
+  signUpWithEmailAndPassword,
+} from "../services/userApi";
+import { useSnackbars } from "./SnackbarProvider";
+
+export type Role = "admin" | "premium" | "classic";
+export type User = {
+  id: string;
+  name: string;
+  email: string;
+  role: Role;
+  isEmailVerified: boolean;
+  lastLogin: Date;
+};
+
+type AuthStatus = "loading" | "connected" | "not-connected" | "error";
```

Comment:
Ton status error n'est jamais utilisé => en cas d'erreur tu va dans l'état "not-connected"